sources/typesafe-config/src/main/java/io/smallrye/configsource/TypeSafeConfigConfigSource.java

```diff
@@ -0,0 +1,53 @@
+package io.smallrye.configsource;
+
+import java.io.Serializable;
+import java.util.Collections;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+import org.eclipse.microprofile.config.spi.ConfigSource;
+
+import com.typesafe.config.Config;
+import com.typesafe.config.ConfigFactory;
+
+public class TypeSafeConfigConfigSource implements ConfigSource, Serializable {
+
+    static final int DEFAULT_ORDINAL = 50;
+
+    private final Map<String, String> properties;
+    private final int ordinal;
+
+    public TypeSafeConfigConfigSource() {
+        final Config config = ConfigFactory.load();
```

Comment:
I'm not very familiar with TypeSafe Config.

Is there any advantage to loading the properties at init vs just keeping the `Config` instance?

Reply:
When calling `config.entrySet()`, the config is looping over all the properties. I'm thinking it's better to do it once and then rely on the performance of a `map.getValue()`.